src/routes/UserService/userService.controller.ts

```diff
@@ -0,0 +1,84 @@
+import { UserService } from "../../entity/UserService";
+import { AppDataSource } from "../../migration/data-source";
+import { User } from "../../entity/User";
+import { Service } from "../../entity/Service";
+import { Request, Response } from "express";
+
+export const addServiceToUser = async (
+    req: Request,
+    res: Response
+): Promise<void> => {
+    try {
+        const userId = req.params.userId;
+        const serviceId = req.params.serviceId;
+        const user = await AppDataSource.manager.findOneBy(User, {
+            id: userId,
+        });
+        if (!user) {
+            res.json({
+                message: "User not found",
+                status: 404,
+            });
+            return; // this exit the function early because the goal of finding the user failed in the first place
+        }
+
+        const service = await AppDataSource.manager.findOneBy(Service, {
+            id: serviceId,
+        });
+        if (!service) {
+            res.json({
+                message: "Service not found",
+                status: 404,
+            });
+            return; // this exit the function early because the goal of finding the service failed in the first place
+        }
+
+        const userService = new UserService();
+        userService.userId = user.id;
+        userService.serviceId = service.id;
+        await AppDataSource.manager.save(userService);
+        res.json({
+            message: "Service added to user successfully",
+            status: 200,
+        });
+    } catch (error) {
+        res.json({
+            message: error,
+            status: 500,
+        });
+    }
+};
+
+export const getUsersService = async (
+    req: Request,
+    res: Response
+): Promise<void> => {
+    try {
+        const userId = req.params.userId;
+        const user = await AppDataSource.manager.findOneBy(User, {
+            id: userId,
+        });
+        if (!user) {
+            res.json({
+                message: "User not found",
+                status: 404,
+            });
+            return; // this exit the function early because the goal of finding the user failed in the first place
```

Comment:
use query builder to get users with the service. reference getUserWithRole() in user. controller.ts to see how it is done